pyvista/core/pointset.py

```diff
@@ -1280,11 +1280,18 @@ def plot_curvature(self, curv_type='mean', **kwargs):
     def volume(self) -> float:
         """Compute the volume of the point grid.
 
-        This extracts the external surface and computes the interior
-        volume.
+        .. versionchanged:: 0.37.0
+           This now calculates the sum of the volume of all cells.
+
         """
-        surf = self.extract_surface().triangulate()
-        return surf.volume
+        vols = self.compute_cell_sizes(length=False, area=False, volume=True)
+        return np.sum(vols["Volume"])
+
+    @property
+    def area(self) -> float:
+        """Compute the area of the point grid if 2D."""
```

Comment:
This method wasn't clear to me on multiple points, but admittedly because I also hadn't realised how `compute_cell_sizes` worked.

1. It wasn't clear what "if 2D" meant: how would it behave if the grid is not 2D? I now know that we just get zero.
2. When talking about "area" I would have assumed (coming from a `PolyData` mindset) that this would also work for space-filling grids, by extracting the surface. I see that this is not the case. Do you think this would make sense? It might make the semantics harder to argue about ("if 2D grid, gives you the total cell area, otherwise the area of the extracted surface"). Especially since for a 2d grid we only count one "side", whereas for a 3d grid we'd have all sides included.
3. It might help clear some of the confusion by adding `Returns` and `Examples` blocks to these docstrings. Maybe show the volume/area of the same 2D and 3D grids. That might prevent some user confusion.

Reply:
Please check new docstring.  PolyData is the odd type out here.  I also added a note about how to obtain the outer surface area of a mesh with 3D cells.